### ISSUE TITLE
bindfs + php-fpm returns file not found.

### ISSUE THREAD

**Issue by canfone** [mpartel/bindfs#44]
I'm trying to setup my local web server using vagrant. My vagrant shared folder is in my home folder (~/home/vagrant/www) and I want to use bindfs to mount this folder inside /var/www.

These are the specs of my virtual machine:

Apache/2.4.23 (Ubuntu)
PHP 7.0.12
Ubuntu 14.04
I am using php-fpm to execute php scripts but after using bindfs, my site will always return File not found.

Also here is my virtualhost configuration:

```
<VirtualHost *:80>
   ServerName project1.dev

    ## Vhost docroot
    DocumentRoot "/var/www/project1/public"

    ## Directories, there should at least be a declaration for /var/www/project1/public

   <Directory "/var/www/project1/public">
       Options Indexes FollowSymlinks MultiViews
       AllowOverride All
       Require all granted

       <FilesMatch "\.php$">
          Require all granted
          SetHandler proxy:fcgi://127.0.0.1:9000
       </FilesMatch>

   </Directory>

   ## Logging
   ErrorLog "/var/log/apache2/av_anhk5lpgjldb_error.log"
   ServerSignature Off
   CustomLog "/var/log/apache2/av_anhk5lpgjldb_access.log" combined

   ## Server aliases 
   ServerAlias www.project1.dev

   ## SetEnv/SetEnvIf for environment variables
   SetEnv APP_ENV dev
   SetEnvIf Authorization "(.*)" HTTP_AUTHORIZATION=$1

   ## Custom fragment
</VirtualHost>
```

Anyone can help me?

**Comment by mpartel** [comment]
- Is `user_allow_other` enabled in `/etc/fuse.conf`?
- Which version of bindfs are you using? Versions 1.13.1-1.13.3 have an unfortunate bug in them that could cause this.
- What does your bindfs command look like?

**Comment by canfone** [comment]
Hi @mpartel ,

Thanks for your response. I am using bindfs version 1.12.3 and currently my `user_allow_other` is commented. Could that be the reason? My bindfs command is this `sudo bindfs -o perms=0750,mirror-only=www-data,force-group=www-data,force-user=www-data /home/vagrant/www /var/www`

**Comment by mpartel** [comment]
Yes, you need `user_allow_other` uncommented unless PHP-FPM runs as the
same user as bindfs.

On 9 Nov 2016 7:35 am, "Cliff Richard Anfone" <notifications@github.com>
wrote:

Hi @mpartel <https://github.com/mpartel> ,

Thanks for your response. I am using bindfs version 1.12.3 and currently my
user_allow_other is commented. Could that be the reason? My bindfs command
is this sudo bindfs -o
perms=0750,mirror-only=www-data,force-group=www-data,force-user=www-data
/home/vagrant/www /var/www

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/mpartel/bindfs/issues/44#issuecomment-259350809>, or mute
the thread

**Comment by canfone** [comment]
Hi @mpartel ,

I already uncommented the `user_allow_other` in my `/etc/fuse.conf` but my virtualhost is still returning File not found. What could be the other possible reason to this?

**Comment by mpartel** [comment]
Hmm..
Is PHP-FPM running as `www-data`?
Do you see the files when you `ls /var/www`?
Anything interesting in the Apache logs?

You can also try running bindfs with `-d` to get (lots of) debug output.

**Comment by canfone** [comment]
This is what I'll see if I run this command.
```
$ ps aux | grep php-fpm
108:root      1073  0.0  3.0 318612 31240 ?        Ss   08:18   0:00 php-fpm: master process (/etc/php/7.0/fpm/php-fpm.conf)
114:www-user  1149  0.0  0.8 318612  8636 ?        S    08:18   0:00 php-fpm: pool www
115:www-user  1150  0.0  0.8 318612  8636 ?        S    08:18   0:00 php-fpm: pool www
```

I can also see my files if I run `sudo ls /var/www` so I'm not sure what did I miss.

**Comment by canfone** [comment]
Hi @mpartel,

Thank you so much for your help. I manage to successfully run php-fpm + bindfs in my virtual machine. I just made sure that user who is running php-fpm and apache are the one I set in my bindfs command. My apache is run by `www-user` so I change my command to `sudo bindfs -o perms=0755,mirror-only=www-user,force-group=www-data,force-user=www-user /home/vagrant/www /var/www` and made sure that apache is also run by `www-user`.

Once again thanks for your help.
